### DISCUSSION TITLE
Websocket connection closed immediately after connected

### DISCUSSION THREAD

**Discussion by tanto-satu** [featbit/featbit#660] [Q&A]
Hi,

Currently im installing featbit in my cloud server in GCP using GCE, but after installing with docker compose, and setup the nginx, when trying to connect from postman, i got this error

<img width="1090" alt="Screenshot 2024-09-11 at 17 05 15" src="https://github.com/user-attachments/assets/82e0cb54-4a21-4575-8f7b-cbe336552b1f">

is it expected?

and also when trying to using react SDK from featbit, in several machine the flag is not stored in local storage, but when i see the network tab, the connection is returning 101 from /streaming URL, but there is no receive message from server, only push message from client is available there. is anyone have the same experience?

**Comment by deleteLater** [comment]
Hi @tanto-satu, sorry for the late reply. The server returns 4003 error when we failed to validate the websocket connection. In your case, I'm gonna assume that you got this request string `wss://app-eval...` from our SDK, but this token would expire in 30 seconds, which is most likely the reason why the server rejected it.

> in several machine the flag is not stored in local storage
> there is no receive message from server

If this problem is caused by the 4003 error, I would recommend you try to check if your server's local time is correctly synchronized.

**Relevant code snippet**

- https://github.com/featbit/featbit/blob/main/modules/evaluation-server/src/Streaming/StreamingMiddleware.cs#L36-L46
- https://github.com/featbit/featbit/blob/main/modules/evaluation-server/src/Streaming/Connections/RequestValidator.cs#L25-L31
